cli/azd/resources/apphost/templates/main.bicept

```diff
@@ -112,4 +115,12 @@ output SERVICE_BINDING_{{alphaSnakeUpper $name}}_NAME string = resources.outputs
 {{- range $param, $value := .OutputParameters}}
 output {{bicepParameterName $param}} {{$value.Type}} = {{bicepParameterName $value.Value}}
 {{- end -}}
+{{if .RequiresStorageVolume -}}
+output AZD_OPS_BIND_MOUNT_ACCOUNT string = resources.outputs.AZD_OPS_BIND_MOUNT_ACCOUNT
```

Comment:
yes, one Storage account for all the volumes in the app.  Each volume uses one file-share inside that storage account.

Reply:
The `AZD_OPS_` is a way to introduce "operations" that azd runs after provisioning.  Some sort of built-in hooks.
This really becomes a azd-specific command; that's why I tried this.  But, we can talk about it

Reply:
Could we ensure that we're consistent when we reference volumes and bind mounts? e.g. "volumes" and "bind mounts" are both "mounts" so if we're naming something that facilitates both, use just "mounts". So if we have one app that uses both bind mounts and volumes, it will share a single storage account for all mounts.

Reply:
>  it will share a single storage account for all mounts.

That's how it works.  There is just one Storage account for all volumes and one FileShare for each volume.
The only difference for bindMount is that it is ingested with files from the device running azd

Reply:
Sorry, just to press on the point a bit, is it one storage account shared between volumes **and** bind mounts? This is why the terminology is helpful in my mind :smile:

Reply:
yes, one single storage account for both.
For simple volumes, the volume description in Aspire-manifest provides a `name`, like:

```json
      "volumes": [
        {
          "name": "qdrant-data",
          "target": "/qdrant/storage",
          "readOnly": false
        }
      ],
```

So, we use the `name` to create the fileShare in the storage account with a name like `${resourcName}${VolumeName}FileShare`  .  Assuming the resource name is unique, this name should never crash with some other volume within the same app.

Then, for mountBinds, the aspire-manifest does not give us a name, like:

```json
      "bindMounts": [
        {
          "source": "../MySql.ApiService/data",
          "target": "/docker-entrypoint-initdb.d",
          "readOnly": false
        }
      ],
```

So, azd is generaing a `name` for each one using `${resourceName}Bm${incrementalCounter}FileShare`
the `incrementalCounter` is the index for iterating the bindMounts array, allowing one resource to have a unique name for each bindMount which is used to create a FileShare next to the other volumes.

If you look at the storage account for one application with volumes and with bindMounts, you would see the two naming patterns for each file-share.